test/server/http_time_tracking_filter_integration_test.cc

```diff
@@ -37,6 +39,22 @@ class HttpTimeTrackingIntegrationTest
 INSTANTIATE_TEST_SUITE_P(IpVersions, HttpTimeTrackingIntegrationTest,
                          testing::ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()));
 
+TEST_P(HttpTimeTrackingIntegrationTest,
+       DiesWhenBothEnvoyApiV2AndV3ResponseHeadersAreSetInConfiguration) {
+  const std::string invalid_configuration = R"EOF(
+  name: time-tracking
+  typed_config:
+    "@type": type.googleapis.com/nighthawk.server.ResponseOptions
+    response_headers:
```

Comment:
Here a user needs to put the exact same configuration contents into a field with a different name. 
I wonder, will protocol buffers even be able to distinct between the two types as it deserializes them? If it does, can we bully it into mechanically converting it, based on our knowledge that the messages are actually defined identically?

Reply:
Thank you for the offline discussion. The summarize the conclusion - considering our goal is to allow users to use v3 if they want to, performing any sort of serialization / deserialization trick wouldn't achieve it. On top of that even if the discussed approach would work, it doesn't follow the good practices set for proto messages which might cause unexpected results in the future.